static/app/components/keyValueData/index.tsx

```diff
@@ -78,7 +85,7 @@ export function Content({
   );
 
   return (
-    <ContentWrapper hasErrors={hasErrors} {...props}>
+    <ContentWrapper hasErrors={Boolean(hasErrors || isSuspectFlag)} {...props}>
```

Comment:
tbh, this component [does support custom value nodes](https://sentry.sentry.io/stories/?name=app/components/keyValueData/index.stories.tsx) (see img below) so i'm surprised we've hard-coded things like `hasErrors` here and `{hasErrors && <AnnotatedTextErrors errors={errors} />}` below. We could use a custom keyValue item for errors (which is a refactor now at this point) and also for this new suspect thing: 
```
return flags.map(f => {
  return {
    item: {
      key: f.flag,
      subject: f.flag,
      value: suspectFlagNames.includes(f.flag)
        ? (
          <span style={{color: theme.red300}}>
            {f.result.toString()}
            <StyledQuestionTooltip size="xs" title={t('Suspect Flag')} />
          </span>
        ) : f.result.toString(),
    },
  };
});
```



![SCR-20241029-nduv](https://github.com/user-attachments/assets/fc40c151-e95b-4fbd-b9ad-60e5a3aee176)

Reply:
the errors one is a bit more complicated, i'm not 100% sure how it works so don't want to touch it in this PR, but i can update the suspect tooltip

Reply:
fair to put a refactor like that in a 2nd PR, today or later far-far-future